### ISSUE TITLE
Decay scheduler based on validation score

### ISSUE THREAD

**Issue by cccs-km** [OpenNMT/OpenNMT-tf#933]
Hi All,
I've been looking through the documentation and I am looking for a decay strategy that is implemented based off of the validation scores. I've seen this: [https://opennmt.net/OpenNMT/training/decay/](https://opennmt.net/OpenNMT/training/decay/), but nothing equivalent in the opennmt-tf documentation. Are there parameters I can set with one of the schedulers in the opennmt-tf that would emulate this?

If any pieces of my config are needed let me know :)

**Comment by guillaumekln** [comment]
Hi,

Currently there is no learning rate scheduler that is based on the validation score.

You can probably manage this logic with a script that runs one training epoch, evaluates the model, update the learning rate configuration accordingly, and then repeats.

See this [example](https://opennmt.net/OpenNMT-tf/faq.html#how-to-count-the-number-of-epochs) to run only one training epoch. The evaluation can be run using the `eval` subcommand of the `onmt-main` entrypoint. It writes the evaluation scores on the standard output.

**Comment by cccs-km** [comment]
Are there plans to include a learning rate scheduler like that in opennmt-tf?

And if not, thank you for the info. I'll look into creating a custom script to handle this!

**Comment by guillaumekln** [comment]
> Are there plans to include a learning rate scheduler like that in opennmt-tf?

There are no plans to add such schedulers at the moment.

**Comment by cccs-km** [comment]
Fair, thank you again for all your help!
